websites/api.py

```diff
@@ -224,3 +228,18 @@ def detect_mime_type(uploaded_file: UploadedFile) -> str:
     magic = Magic(mime=True)
     chunk = next(uploaded_file.chunks(chunk_size=2048))
     return magic.from_buffer(chunk)
+
+
+def reset_publishing_fields(website_name: str):
```

Comment:
Should we also use this function in the `publish` and `preview` `WebsiteViewSet` methods?